core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -512,10 +512,12 @@ protected ExtractionResult visitComparisonExpression(ComparisonExpression node,
                 Type castTargetType = requireNonNull(expressionTypes.get(NodeRef.of(castExpression)), "No type for Cast target expression");
                 if (castSourceType instanceof VarcharType && castTargetType == DATE && !castExpression.isSafe()) {
                     Optional<ExtractionResult> result = createVarcharCastToDateComparisonExtractionResult(
-                            node,
+                            castExpression,
+                            normalized.getComparisonOperator(),
                             (VarcharType) castSourceType,
                             normalized.getValue(),
-                            complement);
+                            complement,
+                            node);
```

Comment:
Why not `normalized` here?

Reply:
You mean instead of passing the fields separately? Yes, that's a good idea.

Reply:
No, i meant passing `normalized` instead of `node` for "original node" (of course, name to be changed).
This way, if we have "remaining expression", this is a normalized one, instead of original.

Nit anyway.

Reply:
Got it. Let me think about it. I'll post a follow-up.